### ISSUE TITLE
race condition with loading nvim settings and using the settings

### ISSUE THREAD

**Issue by correabuscar** [neovide/neovide#1590]
**Describe the bug**
If there's a long enough delay in loading the neovide settings from nvim such as `let g:neovide_transparency=0.8` then whatever the [defaults](https://github.com/neovide/neovide/blob/0b6e20c1994c761814eb6a8ad33dea90ade97573/src/window/settings.rs#L20-L35) are for it will be used instead.

**To Reproduce**
1. Apply this patch:
```patch
diff --git a/src/bridge/mod.rs b/src/bridge/mod.rs
index 89afd63..e0b73f0 100644
--- a/src/bridge/mod.rs
+++ b/src/bridge/mod.rs
@@ -85,7 +85,12 @@ async fn start_neovim_runtime() {
     let nvim = Arc::new(nvim);
 
     start_ui_command_handler(nvim.clone());
+    use std::{thread,time};
+    log::info!("!!! waiting before reading settings!");
+    thread::sleep(time::Duration::from_millis(2000));
+    log::info!("!!! reading settings now!");
     SETTINGS.read_initial_values(&nvim).await;
+    log::info!("!!! riv done!");
     SETTINGS.setup_changed_listeners(&nvim).await;
 
     match io_handler.await {
```
2. set `let g:neovide_transparency=0.8` in ~/.config/nvim/init.vim
3. start neovide
4. see that (even 5 seconds later), the window is still fully opaque and not 20% transparent as per the setting
5. can press `a` to append to see half the window is now 20% transparent
6. or can `:q` to see the bottom part of the window become transparent

The problem is, that depending on system load, sometimes you get the settings applied, sometimes not.

**Expected behavior**
apply the settings consistently, or if not right from the start then (2 seconds) later (whenever they're finally read)

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Gentoo amd64
 - Neovide Version 0.10.1, commit 653fd71f78e136df8b9f9175157170781b2e7c74 (HEAD -> main, origin/main, origin/HEAD) Date:   Tue Oct 4 00:26:48 2022 +1100
 - Neovim Version v0.8.0

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

**Additional context**
https://github.com/neovide/neovide/issues/1580#issuecomment-1272063568

**Comment by s-cerevisiae** [comment]
I don't know this can be easily fixed, but I have a workaround for it:
Use Neovim option `--cmd` to ensure Neovide settings are loaded before any config.
```
neovide -- --cmd "<nvim command to set the config>" --cmd "<another one>"
```

**Comment by fredizzimo** [comment]
I think we have fixed this issue in Neovim 0.13.0 or maybe even in some of the 0.12.x releases, the settings loading system was overhauled.
